### PR TITLE
Update behavior for UpdateHash

### PR DESCRIPTION
**Pull request by SNiukalov**
Fixes #[FORDTCN-7369](https://adc.luxoft.com/jira/browse/FORDTCN-7369)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
Avoid multiple calls UpdateHash during resumption

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc [hunk 1]
```diff
@@ -229,6 +226,22 @@ std::set<ModuleUid> RCAppExtension::InteriorVehicleDataSubscriptions() const {
   return subscribed_interior_vehicle_data_;
 }
 
+bool RCAppExtension::AddPendingSubscription(const ModuleUid& module) {
+  return pending_subscriptions_.insert(module).second;
+}
+
+void RCAppExtension::RemovePendingSubscription(const ModuleUid& module) {
+  subscribed_interior_vehicle_data_.erase(module);
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc:234]: @SNiukalov why it called _RemovePendingSubscription_ but we removing from _subscribed_interior_vehicle_data__?
shouldn't it be pending_subscriptions_.erase(module); ?

**Review by YarikMamykin — APPROVED** [review]

**Review by LuxoftAKutsan — APPROVED** [review]

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc (hunk 1 above)
Comment (SNiukalov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_app_extension.cc]: @LitvinenkoIra 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/129/commits/76ecf5e3dcc3795b8559ff62193dbf46fc2bdba7

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_app_extension.h [hunk 2]
```diff
@@ -182,6 +182,36 @@ class RCAppExtension : public application_manager::AppExtension {
    */
   std::set<ModuleUid> InteriorVehicleDataSubscriptions() const;
 
+  /**
+   * @brief AddPendingSubscription add pending subscription
+   * @param module interior data specification(zone, data type)
+   * @return
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_app_extension.h:188]: @SNiukalov please specify the return value

**Review by LitvinenkoIra — APPROVED** [review]

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_app_extension.h (hunk 2 above)
Comment (SNiukalov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_app_extension.h]: @LitvinenkoIra 
Thanks, fixed in:https://github.com/LuxoftSDL/sdl_core/pull/129/commits/6f71dec232f0b88b3beeacc1c02fdc23af9d6190

**Review by IGapchuk — APPROVED** [review]

**Pull request merged**